app/controllers/application_controller.rb

```diff
@@ -68,11 +68,16 @@ def find_platform(param = :id)
     @platform_name = @platform.formatted_name
   end
 
+  def query_project(platform, name)
```

Comment:
From a rails-y perspective, this feels like it should live somewhere else as it's not setting a class variable. I wonder if it should just be a class method in the `Project` model

Reply:
I'm struggling to know where to put this and what to call it when it's there; feel like I don't know Rails well enough. The goal of course is to factor it out so I can use it in `/dependencies_bulk` which doesn't set the project class variable. I think the way I factored it out is the smallest change to avoid cut-and-paste, but not necessarily the right factoring or reorganization.

Reply:
I'd suggest in `app/models/project.rb` creating a method `self.find_with_versions(platform, name)` that has this code and then you can just call `Project.find_with_versions(platform, name)` where it's needed